sympy/stats/stochastic_process_types.py

```diff
@@ -224,12 +234,17 @@ def joint_distribution(self, *args):
                 raise ValueError("Expected a RandomIndexedSymbol or "
                                 "key not  %s"%(type(arg)))
 
-        if args[0].pspace.distribution == None: # checks if there is any distribution available
+        t = Dummy('t')
+        dist = self.distribution(t)
+        if type(dist) is Distribution: # checks if there is any distribution available
```

Comment:
`isinstance` did not work here as all other Distributions are also derived from `Distribution`, so had to use this.

Reply:
AFAICT, this was needed to return an un-evaluated `JointDistribution` object if there isn't any associated distribution for the `RandomIndexedSymbol`. However, now since, we have introduced abstract `Distribution` class, we don't need this check, since both, L241 and L244 will fail and hence an evaluated `JointDistribution` object can be returned as default.

Reply:
`args[0].pspace.distribution` is being set to None by default due to the changes in code, so I had to check `args[0].pspace.process` for the existence of `density` instead (`DiscreteMarkovChain` do not have density, so cannot call without if else).